cwms-data-api/src/main/java/cwms/cda/data/dao/StreamDao.java

```diff
@@ -1,130 +1,257 @@
+    static final int STREAM_FLOWS_INTO_STATION_COLUMN_INDEX = 5;
+    static final int STREAM_FLOWS_INTO_BANK_COLUMN_INDEX = 6;
+    static final int STREAM_DIVERTS_FROM_STREAM_COLUMN_INDEX = 7;
+    static final int STREAM_DIVERTS_FROM_STATION_COLUMN_INDEX = 8;
+    static final int STREAM_DIVERTS_FROM_BANK_COLUMN_INDEX = 9;
+    static final int STREAM_STREAM_LENGTH_COLUMN_INDEX = 10;
+    static final int STREAM_AVERAGE_SLOPE_COLUMN_INDEX = 11;
+    static final int STREAM_COMMENTS_COLUMN_INDEX = 12;
+    private static final String DB_STATION_UNITS = "km";
+    public static final String DB_STREAM_SLOPE_UNITS = "%";
 
-public class StreamDao extends JooqDao<Stream> {
     public StreamDao(DSLContext dsl) {
         super(dsl);
     }
 
-    public Stream getStream(String streamId, String unitSystem, String officeId) {
-        String pStationUnit = UnitSystem.EN.value().equals(unitSystem) ? Unit.MILE.getValue() :
-                Unit.KILOMETER.getValue();
-        CwmsDbStreamJooq streamJooq = new CwmsDbStreamJooq();
-
-        return connectionResult(dsl, c -> {
-            StreamT streamResult = streamJooq.retrieveStreamF(c, streamId, pStationUnit, officeId);
-            return new Stream.Builder(streamId, streamResult.getStartsDownstream(),
-                    streamResult.getLength(), streamResult.getOfficeId())
-                    .withDivertingStreamId(streamResult.getDivertsFromStream())
-                    .withDiversionStation(streamResult.getDivertsFromStation())
-                    .withDiversionBank(streamResult.getDivertsFromBank())
-                    .withReceivingStreamId(streamResult.getFlowsIntoStream())
-                    .withConfluenceStation(streamResult.getFlowsIntoStation())
-                    .withConfluenceBank(streamResult.getFlowsIntoBank())
-                    .withComment(streamResult.getComments())
-                    .withAverageSlope(streamResult.getAverageSlope())
-                    .withStreamLocations(getStreamLocationsOnStream(streamId, unitSystem, officeId))
-                    .withTributaries(getTributaries(streamId, unitSystem, officeId))
-                    .withStreamReaches(getReaches(streamId, officeId))
-                    .build();
+    /**
+     * Retrieve a list of streams
+     * @param officeIdMask - the office id mask
+     * @param streamIdMask - the stream id mask
+     * @return a list of streams
+     */
+    public List<Stream> retrieveStreams(String officeIdMask, String streamIdMask) {
+        return connectionResult(dsl, conn -> {
+            setOffice(conn, officeIdMask);
+            ResultSet streams = CWMS_STREAM_PACKAGE.call_CAT_STREAMS(DSL.using(conn).configuration(), streamIdMask,
+                    DB_STATION_UNITS, null, null, null,
+                    null, null, null, null,
+                    null, null, null, null, null,
+                    null, null, officeIdMask)
+                    .intoResultSet();
```

Comment:
Why are you turning into a ResultSet instead of using the jOOQ Result class?

Reply:
I saw hec-cwms-data-access did something similar so i followed suit. I replaced with Result, similar to the PoolsDao.